bellatrix.playwright/src/main/java/solutions/bellatrix/playwright/components/common/validate/Validator.java

```diff
@@ -0,0 +1,91 @@
+import solutions.bellatrix.core.configuration.ConfigurationService;
+import solutions.bellatrix.playwright.components.WebComponent;
+import solutions.bellatrix.playwright.configuration.TimeoutSettings;
+import solutions.bellatrix.playwright.configuration.WebSettings;
+import solutions.bellatrix.playwright.services.WebService;
+import solutions.bellatrix.playwright.utilities.DebugLogger;
+import solutions.bellatrix.playwright.utilities.functionalinterfaces.AssertionMethod;
+import solutions.bellatrix.playwright.utilities.functionalinterfaces.ComparatorMethod;
+
+import java.util.function.Supplier;
+
+@UtilityClass
+public class Validator extends WebService {
+    private static final TimeoutSettings timeoutSettings = ConfigurationService.get(WebSettings.class).getTimeoutSettings();
+
+    public static void validateCondition(ComparatorMethod condition) {
+        boolean isConditionMet = false;
+
+        long pollingInterval = timeoutSettings.inMilliseconds().getSleepInterval();
+        long timeout = timeoutSettings.inMilliseconds().getValidationsTimeout();
+
+        long startTime = System.currentTimeMillis();
+
+        while (!isConditionMet && (System.currentTimeMillis() - startTime) <= timeout) {
+            try {
+                if (condition.evaluate()) {
+                    isConditionMet = true;
+                } else {
+                    Thread.sleep(pollingInterval);
+                }
+            } catch (Exception e) {
+                // ignore
+            }
+        }
+
+        if (!isConditionMet) {
+            throw new AssertionFailedError(String.format("Assertion failed.\n%s", condition.toString()));
+        }
+    }
+
+    public static void validateCondition(ComparatorMethod condition, long timeoutInMilliseconds, long pollingIntervalInMilliseconds) {
+        boolean isConditionMet = false;
+
+        long startTime = System.currentTimeMillis();
+
+        while (!isConditionMet && (System.currentTimeMillis() - startTime) <= (timeoutInMilliseconds)) {
+            try {
+                if (condition.evaluate()) {
+                    isConditionMet = true;
+                } else {
+                    Thread.sleep(pollingIntervalInMilliseconds);
+                }
+            } catch (Exception e) {
+                // ignore
+            }
+        }
+
+        if (!isConditionMet) {
+            throw new AssertionFailedError(String.format("Assertion failed.\n%s", condition.toString()));
+        }
+    }
+
+    public static <T, V> void validate(AssertionMethod assertion, WebComponent component, String attributeName, V value, Supplier<T> supplier, String prefix) {
+        var error = String.format("\u001B[0mThe %s of \u001B[1m%s \u001B[2m(%s)\u001B[0m%n" +
+                        "  Should %s: \"\u001B[1m%s\u001B[0m\"%n" +
+                        "  %" + prefix.length() + "sBut was: \"\u001B[1m%s\u001B[0m\"%n" +
+                        "Test failed on URL: \u001B[1m%s\u001B[0m",
```

Comment:
these '001B[0m' symbols are for the coloring of the output, and they are not looking good in Jenkins. Not sure if we should keep them or maybe add some config option whether to color the output or not.

Reply:
I'll remove the coloring from this output.